ssh/client.go

```diff
@@ -1,53 +1,119 @@
 package ssh
 
 import (
-	"bytes"
-	"errors"
 	"fmt"
-	"io"
 	"io/ioutil"
 	"os"
+	"os/exec"
 
 	"github.com/docker/docker/pkg/term"
 	"github.com/docker/machine/log"
+	"github.com/docker/machine/utils"
 	"golang.org/x/crypto/ssh"
 )
 
-type Client struct {
-	Config   *ssh.ClientConfig
+type Client interface {
+	Output(command string) (string, error)
+	Shell() error
+}
+
+type ShelloutClient struct {
+	BaseArgs   []string
+	BinaryPath string
+}
+
+type GoNativeClient struct {
+	Config   ssh.ClientConfig
 	Hostname string
 	Port     int
 }
 
+type Auth struct {
+	Passwords []string
+	Keys      []string
+}
+
+type SSHClientType string
+
 const (
 	maxDialAttempts = 10
 )
 
-func NewClient(user string, host string, port int, auth *Auth) (*Client, error) {
-	config, err := NewConfig(user, auth)
+const (
+	NoDefault SSHClientType = ""
+	Shellout  SSHClientType = "shellout"
+	GoNative  SSHClientType = "gonative"
+)
+
+var (
+	baseSSHArgs = []string{
+		"-o", "IdentitiesOnly=yes",
+		"-o", "StrictHostKeyChecking=no",
```

Comment:
Yeah.  It's known issue unfortunately: https://github.com/docker/machine/issues/534

Hopefully we can come up with a better solution in the future.